internal/signalio/json_test.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2022 Criticality Score Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package signalio
+
+import (
+	"encoding/json"
+	"io"
+	"testing"
+
+	"github.com/ossf/criticality_score/internal/collector/signal"
+)
+
+type testJSONWriterSet struct { //nolint:govet
+	UpdatedCount signal.Field[int]
+	Field        string
+}
+
+func (t testJSONWriterSet) Namespace() signal.Namespace {
+	return "test"
+}
+
+type mockWriterJSON struct{}
+
+func (m *mockWriterJSON) Write(p []byte) (n int, err error) {
+	return 0, nil
+}
```

Comment:
Use a `bytes.Buffer` to avoid needing to implement the `io.Writer` interface.
It also allows you to inspect what was written.